### PR TITLE
[react] change sort icons for entity

### PR DESCRIPTION
**Pull request by solnone**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/react/templates/src/main/webapp/app/entities/_entityFolder/_entityFile.tsx.ejs [hunk 1]
```diff
@@ -306,6 +307,19 @@ export const <%= entityReactName %> = () => {
 <%_ } _%>
   };
 
+  const getSortIconByFieldName = (fieldName: string) => {
+<%_ if (paginationNo) { _%>
+    const { sort: sortFieldName, order } = sortState;
+<%_ } else { _%>
+    const { sort: sortFieldName, order } = paginationState;
+<%_ } _%>
+    return (sortFieldName !== fieldName)
```
Comment (qmonmert) [inline generators/react/templates/src/main/webapp/app/entities/_entityFolder/_entityFile.tsx.ejs:316]: I think that ternary into ternary is not understandable 🙂

**Review by qmonmert — APPROVED** [review]

File: generators/react/templates/src/main/webapp/app/entities/_entityFolder/_entityFile.tsx.ejs [hunk 2]
```diff
@@ -306,6 +307,19 @@ export const <%= entityReactName %> = () => {
 <%_ } _%>
   };
 
+  const getSortIconByFieldName = (fieldName: string) => {
+<%_ if (paginationNo) { _%>
+    const { sort: sortFieldName, order } = sortState;
+<%_ } else { _%>
+    const { sort: sortFieldName, order } = paginationState;
```
Comment (solnone) [inline generators/react/templates/src/main/webapp/app/entities/_entityFolder/_entityFile.tsx.ejs:314]: [Destructuring assignment - JavaScript | MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)
```suggestion
    const sortFieldName = sortState.sort;
    const order = sortState.order;
<%_ } else { _%>
    const sortFieldName = paginationState.sort;
    const order = paginationState.order;
```

File: generators/react/templates/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs [hunk 3]
```diff
@@ -107,6 +108,14 @@ export const UserManagement = () => {
   const users = useAppSelector(state => state.userManagement.users);
   const totalItems = useAppSelector(state => state.userManagement.totalItems);
   const loading = useAppSelector(state => state.userManagement.loading);
+  const getSortIconByFieldName = (fieldName: string) => {
+    const { sort: sortFieldName, order } = pagination;
+    return (sortFieldName !== fieldName)
```
Comment (qmonmert) [inline generators/react/templates/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:113]: you can do like the other template (no double ternary)

**Review by DanielFran — APPROVED** [review]

**Comment by DanielFran** [comment]
Thanks @solnone for the contribution

**Pull request merged**